src/tensors/cpu/ruy_adapter.h

```diff
@@ -0,0 +1,468 @@
+    vst1q_s8(&dst[ 7*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32( vget_low_s32(x3.val[1]),  vget_low_s32(x7.val[1]))));
+
+    vst1q_s8(&dst[ 8*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x0.val[0]), vget_high_s32(x4.val[0]))));
+    vst1q_s8(&dst[ 9*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x1.val[0]), vget_high_s32(x5.val[0]))));
+    vst1q_s8(&dst[10*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x2.val[0]), vget_high_s32(x6.val[0]))));
+    vst1q_s8(&dst[11*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x3.val[0]), vget_high_s32(x7.val[0]))));
+    vst1q_s8(&dst[12*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x0.val[1]), vget_high_s32(x4.val[1]))));
+    vst1q_s8(&dst[13*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x1.val[1]), vget_high_s32(x5.val[1]))));
+    vst1q_s8(&dst[14*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x2.val[1]), vget_high_s32(x6.val[1]))));
+    vst1q_s8(&dst[15*rows + dstRowBegin], vreinterpretq_s8_s32(vcombine_s32(vget_high_s32(x3.val[1]), vget_high_s32(x7.val[1]))));
+
+    // clang-format on
+  }
+
+  struct UnquantizeAndAddBiasAndWrite {
+    UnquantizeAndAddBiasAndWrite(float unquant_multiplier, const float *input_bias_prepared)
+        : unquant_multiplier(unquant_multiplier), input_bias_prepared(input_bias_prepared) {}
+
+    void operator()(const int32_t *input, Index rows_A, Index cols_B, float *output) {
+      // Set all registers in lane from same scalar value.
+      float32x4_t multiplier = vdupq_n_f32(unquant_multiplier);
+      const int32x4_t *Input = reinterpret_cast<const int32x4_t *>(input);
+      const int32x4_t *InputEnd = reinterpret_cast<const int32x4_t *>(input + rows_A * cols_B);
+      float32x4_t *Output = reinterpret_cast<float32x4_t *>(output);
+
+      while(Input != InputEnd) {
+        // Bias cycles every column for addition.
+        const float32x4_t *Bias = reinterpret_cast<const float32x4_t *>(input_bias_prepared);
+
+        // InputEnd needs to be determined to end the while loop below.
+        const int32x4_t *RowEnd = reinterpret_cast<const int32x4_t *>(
+            reinterpret_cast<const int32_t *>(Input) + cols_B);
+
+        while(Input != RowEnd) {
+          // Operation happening for 4-elements together:
+          // output = [int32_t]input * [float]quant_mult + [float]bias;
+          float32x4_t floatInput = vcvtq_f32_s32(*Input++);
+          float32x4_t unquantized = vmulq_f32(floatInput, multiplier);
+          *Output++ = vaddq_f32(unquantized, *Bias++);
+        }
+      }
+    }
+
+    float unquant_multiplier;
+    const float *input_bias_prepared;
+  };
+
+  struct UnquantizeAndWrite {
+    UnquantizeAndWrite(float unquant_multiplier) : unquant_multiplier(unquant_multiplier) {}
+
+    void operator()(const int32_t *input, Index rows_A, Index cols_B, float *output) {
```

Comment:
Where is this function called?  ruy doesn't have callbacks.  intgemm is x86 only.